api/src/main/java/org/apache/iceberg/io/CloseableIterable.java

```diff
@@ -148,8 +143,6 @@ public CloseableIterator<E> iterator() {
 
       private ConcatCloseableIterator(Iterable<CloseableIterable<E>> inputs) {
         this.iterables = inputs.iterator();
-        this.currentIterable = iterables.next();
-        this.currentIterator = currentIterable.iterator();
```

Comment:
Why was it necessary to change how these are initialized? Isn't the problem that `iterator` is called twice on the `iterable` passed to `concat`? If that is avoided, then I think this can stay the same.

Reply:
The init code here already implied that there would be at least one element, so this was necessary to make the code work even with empty iterables